### PR TITLE
Misc select tools and snapping fixes

### PR DESCRIPTION
**Pull request by nyalldawson**
@3nids as discussed, here's some followups for the recent changes to the select by radius tool, and also fixing some missing snap indicators for other tools.

### CODE REVIEW

**Comment by elpaso** [comment]
Nice!

**Review by 3nids — APPROVED** [review]
great job!

File: src/app/qgsmaptoolselectradius.cpp [hunk 1]
```diff
@@ -103,11 +104,9 @@ void QgsDistanceWidget::distanceSpinBoxValueChanged( double distance )
 
 QgsMapToolSelectRadius::QgsMapToolSelectRadius( QgsMapCanvas *canvas )
   : QgsMapTool( canvas )
+  , mSnapIndicator( new QgsSnapIndicator( canvas ) )
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply src/app/qgsmaptoolselectradius.cpp]: Fixed

File: src/app/qgsmaptoolselectradius.h [hunk 2]
```diff
@@ -104,14 +105,15 @@ class APP_EXPORT QgsMapToolSelectRadius : public QgsMapTool
 
     //! used for storing all of the maps point for the polygon
     QgsRubberBand *mRubberBand = nullptr;
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply src/app/qgsmaptoolselectradius.h]: Fixed

**Comment by saberraz** [comment]
IMO, the radius select plus plugin selection is more intuitive:
https://plugins.qgis.org/plugins/qgis-select-by-radius-plus-plugin/

**Comment by nyalldawson** [comment]
@saberraz
I'm not familiar with that plugin, but the description sounds identical to the new core tool behaviour. What's the difference?

**Comment by saberraz** [comment]
It's the same functionality. But mainly how user clicks.

**Pull request merged**